added section explaining path merging of url and base parameters

teetotum opened the pull request:
### Description

Added one new section explaining how url and base paths are merged; illustrated with four examples

### Motivation

The simple but erroneous assumption that url and base are simply concatenated is a mistake easy enough to make.
Even if a developer is conscious about making an assumption and therefore cross checking by viewing the documentation he/she will not find the information on the page currently.
The documentation page should have a section or a note addressing this.

### Additional details

[RFC 3986](https://datatracker.ietf.org/doc/html/rfc3986.html#section-5.2)

### Related issues and pull requests

teetotum committed:
added section explaining path merging of url and base parameters

The path merging behavior was not documented here up until now. The simple but erroneous assumption that url and base are simply concatenated is a mistake easy enough to make. The documentation page should have a section or a note addressing this.

teetotum committed:
Merge branch 'main' into patch-2

teetotum committed:
Merge branch 'main' into patch-2

wbamberg committed:
Make Prettier happy

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

teetotum reviewed (commented):
looks good to me

wbamberg reviewed (commented):
Sorry to be slow reviewing this. I think the content is fine, but we usually don't have ad hoc top-level sections in these docs, so perhaps it would be better to have this as part of the preamble, before "Syntax".

It might also be worth having an example like:

```
const E = new URL("../articles", "https://developer.mozilla.org/api/v1/");
// => 'https://developer.mozilla.org/api/articles'
```

?

teetotum committed:
added examples for './path' and '../path'

teetotum committed:
Update files/en-us/web/api/url/url/index.md

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>

teetotum committed:
moved text into a **Note:** under ##Syntax -> ### Parameters

teetotum commented:
I added your example to show the effect of `../articles` and also another one to show the effect of `./articles` to have a complete collection of all combinations (can you confirm? did I miss one?).

I tried your suggestion of moving the text into the preamble. But it felt strange because at that point in the doc the constructor arguments have yet to be introduced to the reader.

Instead I moved the text into a _note_ under _Syntax -> Arguments_ where it seems to fit. I moved the examples into the _Examples_ section but kept them in their own block.

Not sure if the wording in the _note_ is too repetitive and should be trimmed down a bit. What do you think?

wbamberg commented on files/en-us/web/api/url/url/index.md:
```diff
@@ -41,6 +41,14 @@ new URL(url, base)
+> Therefore a trailing slash in `base` or a leading slash in `url` affect how the resulting path is constructed.
+> If you need a strict concatenation of the two arguments the `url` must not have a leading slash
+> and the `base` must have a trailing slash.
+> You can find examples for all possible combinations under [Examples - Merging of url and base paths](#merging-of-url-and-base-paths).
```
```suggestion
> See the examples under [Examples - Merging of url and base paths](#merging_of_url_and_base_paths).
```

wbamberg replied:
However I would also not make this a note.

wbamberg reviewed (changes requested):
Thank you for the update, @teetotum !

I would not make this a note, and I made a suggestion to:
- not claim we cover all possibilities, since I'm not sure we do, and not sure it is all that important to do that
- fix the URL fragment in the link

teetotum committed:
Update files/en-us/web/api/url/url/index.md

Co-authored-by: wbamberg <will@bootbonnet.ca>

teetotum committed:
turned the **Note:** into regular document text

teetotum commented:
I applied your fix and turned the _note_ into regular text. It still sits in the _Parameters_ sections under the listed parameters but doesn't have the blue box anymore.

wbamberg reviewed (approved):
👍 thank you @teetotum !
